src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -305,10 +304,6 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
       getWidgetState(uiState, { searchParameters }) {
         const page = searchParameters.page || 0;
 
-        if (!hasShowPrevious || !page) {
```

Comment:
We used to return `uiState` when page is `0` because otherwise `page=1` shows up in the UI state and in the routing, whereas it's the default value. Therefore we should keep this branch with only the `!page` condition.

Reply:
Thanks for the context. I didn't know that. I tested it and confirmed the behavior you said. So I put it back.